R/flatten.R

```diff
@@ -56,7 +56,14 @@ flatten <- function(tbl, groups, wide_cols, sep = "_") {
   uniquifying <- unique(uniquifying)
 
   out <- split(subset(tbl, select = -idx), subset(tbl, select = idx), sep = sep)
-  missing <- setdiff(wide_cols, colnames(tbl))
+  
+  # in original assays there are no columns with SD-related data (with names ending with "_sd")
+  missing <- if (!any(grepl("_sd$", colnames(tbl)))) {
+    setdiff(wide_cols[!grepl("_sd$", wide_cols)], colnames(tbl))
+  # assays with averaged biological replicates might have columns with SD-related data (with names ending with "_sd")
+  } else {
+    setdiff(wide_cols, colnames(tbl))
+  }
```

Comment:
```suggestion
  missing <- setdiff(wide_cols[!grepl("_sd$", wide_cols)], colnames(tbl)
```

is this else required? It seems that this one line should be enough for both cases (or I am wrong)

Reply:
You're right. I went with if/else to make it visible that logic is dependent on the averaging of biological replicates. I will keep the comment and go with the solution you've suggested.